src/audio/pipeline/pipeline-graph.c

```diff
@@ -184,60 +184,15 @@ void pipeline_disconnect(struct comp_dev *comp, struct comp_buffer *buffer, int
 	irq_local_enable(flags);
 }
 
-static int pipeline_comp_free(struct comp_dev *current,
-			      struct comp_buffer *calling_buf,
-			      struct pipeline_walk_context *ctx, int dir)
-{
-	struct pipeline_data *ppl_data = ctx->comp_data;
-	uint32_t flags;
-
-	pipe_dbg(current->pipeline, "pipeline_comp_free(), current->comp.id = %u, dir = %u",
-		 dev_comp_id(current), dir);
-
-	if (!comp_is_single_pipeline(current, ppl_data->start)) {
-		pipe_dbg(current->pipeline, "pipeline_comp_free(), current is from another pipeline");
-		return 0;
-	}
-
-	/* complete component free */
-	current->pipeline = NULL;
-
-	pipeline_for_each_comp(current, ctx, dir);
-
-	/* disconnect source from buffer */
-	irq_local_disable(flags);
-	list_item_del(comp_buffer_list(current, dir));
```

Comment:
and this?

Reply:
buffer_free() is also doing this today, so this will fix the list item removal to one place.